walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -60,50 +52,74 @@
   }
 
   /** 
-   * Returns an ArrayList of all busy TimeRanges.
+   * Returns an ImmutableList of all busy TimeRanges.
    * Check all events for the day and add TimeRanges of events with our mandatory attendees.
   */
-  private ArrayList<TimeRange> findBusyRanges(Collection<String> attendees, Collection<Event> events) {
-    ArrayList<TimeRange> busyRanges = new ArrayList<TimeRange>();
-    for (Event event : events){
-      for (String attendee : attendees) {
-        if (event.getAttendees().contains(attendee)){
-          busyRanges.add(event.getWhen());
-        }      
-      }
+  private ImmutableList<TimeRange> findBusyRanges(Collection<String> attendees, Collection<Event> events) {
+    ArrayList<TimeRange> busyRanges = new ArrayList<>();
+    for (Event event : events) {
+        // Checks if we have any overlap of attendees
+        if (!Collections.disjoint(attendees, event.getAttendees())) {
+            busyRanges.add(event.getWhen());
+        }
     }
```

Comment:
If you feel like learning something new, I recommend trying to do this with java streams. It is often more clear than for loops.

https://stackoverflow.com/questions/44180101/in-java-what-are-the-advantages-of-streams-over-loops
https://docs.oracle.com/javase/8/docs/api/java/util/stream/Stream.html